compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java

```diff
@@ -290,12 +292,23 @@ private void writeFunction(ByteBuf buf, BIRTypeWriter typeWriter, BIRInstruction
             birbuf.writeBoolean(details.send);
         }
 
+        // Write the instruction vs scope table
+        writeScopes(buf, scopebuf, funcInsWriter.getScopeCount());
+
         // Write length of the function body so that it can be skipped easily.
         int length = birbuf.nioBuffer().limit();
         buf.writeLong(length);
         buf.writeBytes(birbuf.nioBuffer().array(), 0, length);
     }
 
+    private void writeScopes(ByteBuf buf, ByteBuf scopebuf, int scopeCount) {
+        int length = scopebuf.nioBuffer().limit();
+        // 4 is the size of int which is the number of scopes that we are going to add to the beginning of the buffer
+        buf.writeLong(length + 4);
+        buf.writeInt(scopeCount);
```

Comment:
I think this count can also be written within intructionWriter as it is the one which calculates the total count.

Reply:
Since I'm writing the scopes to the buffer on the fly, to write the scope count I'll have to start a brand new buffer and write to it. I felt like thats an overkill. WDYT?